src/reports/filters/AppointmentFilter.tsx

```diff
@@ -0,0 +1,90 @@
+import {StyledForm} from "blaise-design-system-react-components";
+import React, {ReactElement} from "react";
+import dateFormatter from "dayjs";
+import utc from "dayjs/plugin/utc";
+import timezone from "dayjs/plugin/timezone";
+import Breadcrumbs from "../../components/Breadcrumbs";
+import AppointmentResourceDaybatchWarning from "../AppointmentResourcePlanning/AppointmentResourceDaybatchWarning";
+
+dateFormatter.extend(utc);
+dateFormatter.extend(timezone);
+
+interface AppointmentFilterPageProps {
+    title: string
+    reportDate: Date
+    setReportDate: (Date: Date) => void
+    surveyTla: string | undefined
+    setSurveyTla: (string: string) => void
+    submitFunction: () => void;
+}
+
+function AppointmentFilter(props: AppointmentFilterPageProps): ReactElement {
+    const {
+        title,
+        reportDate,
+        setReportDate,
+        surveyTla,
+        setSurveyTla,
+        submitFunction
+    } = props;
+
+    async function submitAppointmentFilters(formValues: Record<string, any>, setSubmitting: (isSubmitting: boolean) => void): Promise<void> {
+        setReportDate(formValues["Date"]);
+        setSurveyTla(formValues["Survey TLA"]);
+        setSubmitting(true);
+        submitFunction();
+    }
+
+    const validateDate = (value: string) => {
+        let error;
```

Comment:
This can be simplified to remove `error` altogether:

```typescript
    const validateDate = (value: string) => {
        if (value === "" || value === undefined) {
            return "Enter a date";
        }

        if (value.match(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/) === null) {
            return "Enter a valid date";
        }
    };
```